dns_cli.py

```diff
@@ -265,224 +226,286 @@ def create_hosted_zone(client, domain, start_domain, base_domain):
+
+    # Scan R53 Zone for matching domains and update records if needed.
     hosted_zones = {}
     for hz in response["HostedZones"]:
         hosted_zones[hz["Name"]] = hz
 
-    domain = domain_name[8:]
-    parts = domain.split(".")
+    parts = domain_name.split(".")
     for _ in range(len(parts) - 1):
         subdom = ".".join(parts) + "."
         print(f"searching for {subdom}... ")
 
         if subdom in hosted_zones:
-
             print("Found hosted zone", hosted_zones[subdom]['Name'])
-
             hosted_zone_id = hosted_zones[subdom]["Id"]
-            hosted_zone_conf = hosted_zones[subdom]
 
             # Does record existing
             response = domain_client.list_resource_record_sets(
                 HostedZoneId=hosted_zone_id,
             )
+            check_response(response)
 
             for record in response['ResourceRecordSets']:
-                if domain == record['Name'][:-1]:
+                if domain_name == record['Name'][:-1]:
                     print(f"Record: {record['Name']} found")
                     print(f"is pointing to LB {record['ResourceRecords'][0]['Value']}")
                     if record['ResourceRecords'][0]['Value'] != elb_name:
-                        if click.confirm(f"This doesnt match with the current LB {elb_name}, Do you wish to update the record?"):
-                            record = {"Name": domain, "Type": "CNAME", "TTL": 300, "ResourceRecords": [{"Value": elb_name}]}
+                        if click.confirm(f"This doesnt match with the current LB {elb_name}, \
+                                         Do you wish to update the record?"):
+                            record = {"Name": domain_name, "Type": "CNAME", "TTL": 300,
+                                      "ResourceRecords": [{"Value": elb_name}]}
                             add_records(domain_client, record, hosted_zone_id, "UPSERT")
                     else:
                         print("No need to add as it already exists")
                     exit()
 
-            record = {"Name": domain, "Type": "CNAME", "TTL": 300, "ResourceRecords": [{"Value": elb_name}]}
+            record = {"Name": domain_name, "Type": "CNAME", "TTL": 300, "ResourceRecords": [{"Value": elb_name}]}
 
-            if not click.confirm(f"Creating R53 record: {record['Name']} -> {record['ResourceRecords'][0]['Value']}\nIn Domain: {subdom}\tZone ID: {hosted_zone_id}\nDo you want to continue?"):
+            if not click.confirm(f"Creating R53 record: {record['Name']} -> {record['ResourceRecords'][0]['Value']}\n\
+                                 In Domain: {subdom}\tZone ID: {hosted_zone_id}\nDo you want to continue?"):
                 exit()
             add_records(domain_client, record, hosted_zone_id, "CREATE")
             exit()
 
         parts.pop(0)
 
     else:
-        print(f"No hosted zone found for {domain}")
+        print(f"No hosted zone found for {domain_name}")
         return
 
 
+def run_atest():
```

Comment:
Better to move tests to their own directory and/or own files.

Reply:
I forgot to take out run_atest.  Will remove

Reply:
Thank you! Resolved in https://github.com/uktrade/copilot-tools/pull/29/commits/47f03694754c5665e60347821bc51297092e9e7d.